CS3219-AY2425S1/cs3219-ay2425s1-project-g11
# Add user profile operations and minor fixes

### Description
This PR aims to add user profile operations
- Users are able to edit their email, username and password
- Users are able to delete their own account
- Added password verification api route for `user-service`
  - only users are able to verify their own passwords

https://github.com/user-attachments/assets/789d77a5-4825-41d8-8c15-ad9a7a980816

Additionally fixes
- Fixed error handling for `signin` and `signup`
- Fixed docker compose version warning
- Fixed hot reloading for user account
- Hide `match` button and remove `question` button
- Make docker do isolated package installation. You still have to do `pnpm i` for locally dev anyways but this is just a quality of life improvement

### How to test
- Create a new user (you'll be deleting this user later)
- Go the `/profile`
- Change email
  - It will check for duplicate emails and prevent the email to be changed
- Change username
  - It will check for duplicate username
- Change password
  - It will check the user's current password
  - It will also check if the user's new password is equals to confirm password
 - Delete account

Closes #86 #87 #85
